andela/ah-infinity-stones
# #61966922 user able to favorite article

#### What does this PR do?
- Implement article favourite and Unfavourite functionalities.
#### Description of Task to be completed?
- Implement favourite unfavourited article.
- Implement unfavourited article.
- Implement tests
#### How should this be manually tested?
1. `git clone -b https://github.com/andela/ah-infinity-stones` ft-User-able-to-favorite-article-61966922  and `$ cd ah-infinity-stones`
2. Create and activate virtual env
- `virtualenv  -p python3 venv`
- `$ source venv/bin/activate`
3. `$ pip3 install -r requirements.txt` to install the dependencies
4. Setup environment variables. Copy all from below and paste them into the .env file. Update the values to your choice.
```export DATABASE_USER='ke-c17rn9b4fvh3'
export DATABASE_NAME='articles_db'
export DATABASE_PASSWORD='admin'
export DATABASE_TEST='articles_db_test'
export EMAIL_HOST_USER='domisemak@gmail.com'
export EMAIL_HOST_PASSWORD='Dommy2018'
export SECRET_KEY='7pgozr2jn7zs_o%i8id6=rddie!*0f0qy3$oy$(8231i^4*@u3''
```
Activate the env variables by `$source .env`.
5. Run migrations: `$ python3 manage.py migrate`
6. Run app - `$ python3 manage.py runserver`
7. Acess `http://127.0.0.1:8000/api/articles/<slug>/favourite` through [Postman](https://www.getpostman.com/), provide inputs in the following format and create account.
```
{ "user": {
"username": "username44",
"email": "username@user.com",
"password": "@Password321"	
}
}
```
Check your email and activate account. Use the registration token to acess the following endpoints:
- `http://127.0.0.1:8000/api/articles` - Post an article. 
-  favourite an article
     `http://127.0.0.1:8000/api/articles/<slug>/favourite` -  change to Post on Postman to favorite the article with “favorite”:”True”  as the data.
      `http://127.0.0.1:8000/api/articles/<slug>/favourite` - change to delete on Postman to unfavourite 

8. Run tests` python3 manage.py test authors.apps.articles.tests.test_views.ArticleFavoriteTestCase`
#### What are the relevant pivotal tracker stories?
[#161966922](https://www.pivotaltracker.com/n/projects/2225605)
#### Screenshots
![image](https://user-images.githubusercontent.com/33648833/50233178-ea8b1400-03c3-11e9-9349-294549b6e29a.png)
![image](https://user-images.githubusercontent.com/33648833/50233218-055d8880-03c4-11e9-8803-b7b4e117643f.png)
![image](https://user-images.githubusercontent.com/33648833/50233333-46ee3380-03c4-11e9-9d78-a82f1b3d948c.png)
![image](https://user-images.githubusercontent.com/33648833/50233319-3f2e8f00-03c4-11e9-90a0-61ef110f7abe.png)
